pkg/plugin/plugin.go

```diff
@@ -9,49 +9,37 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/openshift-azure/pkg/api"
+	pluginapi "github.com/openshift/openshift-azure/pkg/api/plugin/api"
 	"github.com/openshift/openshift-azure/pkg/arm"
 	"github.com/openshift/openshift-azure/pkg/cluster"
 	"github.com/openshift/openshift-azure/pkg/config"
 )
 
 type plugin struct {
-	log             *logrus.Entry
-	config          api.PluginConfig
-	clusterUpgrader cluster.Upgrader
-	armGenerator    arm.Generator
-	configGenerator config.Generator
-	apiValidator    *api.Validator
-	adminValidator  *api.Validator
+	log                     *logrus.Entry
+	config                  api.PluginConfig
+	clusterUpgrader         cluster.Upgrader
+	armGenerator            arm.Generator
+	configGenerator         config.Generator
+	apiValidator            *api.Validator
+	adminValidator          *api.Validator
+	pluginTemplateValidator *api.Validator
 }
 
 var _ api.Plugin = &plugin{}
 
 // NewPlugin creates a new plugin instance
 func NewPlugin(log *logrus.Entry, pluginConfig *api.PluginConfig, skipValidate ...bool) (api.Plugin, []error) {
```

Comment:
skipValidate is now defunct - is this expected?

Reply:
yes, we need to remove it. It was added as a temporary workaround

Reply:
are you going to clean this up in this PR or in a follow up? If follow up, mind opening an issue to track?

Reply:
in this one.